examples/jetstream/push-consumer/java/Main.java

```diff
@@ -0,0 +1,198 @@
+            System.out.printf("%d messages queued\n", queuedMsgs);
+
+            // To *receive* a message, call `nextMsg` with a timeout. The timeout
+            // applies when pending count is zero and the consumer has fully caught
+            // up to the available messages in the stream. If no messages become
+            // available, this call will only block until the timeout.
+            Message msg = sub.nextMessage(Duration.ofSeconds(1));
+            System.out.printf("Received '%s'\n", msg.getSubject());
+
+            // By default, the underlying consumer requires explicit acknowledgements,
+            // otherwise messages will get redelivered.
+            msg.ack();
+
+            // Let's receive and ack another.
+            msg = sub.nextMessage(Duration.ofSeconds(1));
+            System.out.printf("Received '%s'\n", msg.getSubject());
+            msg.ack();
+
+            // Checking out our pending information, we see there are no more
+            // than four remaining.
+            queuedMsgs = sub.getPendingMessageCount();
+            System.out.printf("%d messages queued\n", queuedMsgs);
+
+            // Unsubscribing this subscription will result in the ephemeral consumer
+            // being deleted. Note, even if this is omitted and the process ends
+            // or is interrupted, the server will eventually clean-up the ephemeral
+            // when it determines the subscription is no longer active.
+            sub.unsubscribe();
+
+            // ### Durable (AddConsumer)
+            // An explicit and safer way to create consumers is using `jsm.addOrUpdateConsumer`.
+            // For push consumers, we must provide a `DeliverSubject` which is the
+            // subject messages will be published to (pushed) for a subscription to
+            // receive them.
+            //
+            // Important: such a short AckWait should not be used, it's simply for demonstration purposes in this example.
+            System.out.println("\n# Durable (AddConsumer)");
+
+            String consumerName = "handler-1";
+            ConsumerConfiguration cc = ConsumerConfiguration.builder()
+                    .durable(consumerName)
+                    .deliverSubject(consumerName)
+                    .ackPolicy(AckPolicy.Explicit)
+                    .ackWait(Duration.ofSeconds(1))
+                    .build();
+            jsm.addOrUpdateConsumer(streamName, cc);
+
+            // Now that the consumer is created, we need to bind a client subscription
+            // to it which will receive and process the messages. This can be done
+            // using the `PushSubscribeOptions.bind` option which requires the consumer
+            // to have been pre-created. The subject can be omitted since that was
+            // already defined on the consumer. Subscriptions to consumers cannot
+            // independently define their own subject to filter on.
```

Comment:
This might be the place to mention, that if a subscribe subject _is_ provided it _must_ exactly match the filter subject.

Reply:
Done